src/ray/gcs/gcs_client/service_based_gcs_client.cc

```diff
@@ -52,6 +52,32 @@ void ServiceBasedGcsClient::Disconnect() {
   RAY_LOG(INFO) << "ServiceBasedGcsClient Disconnected.";
 }
 
+uint64_t ServiceBasedGcsClient::Reconnect(uint64_t reconnect_count) {
+  absl::MutexLock lock(&mutex_);
+  if (reconnect_count_ >= reconnect_count) {
+    int num_attempts = 0;
+    while (num_attempts < RayConfig::instance().gcs_service_rpc_connect_retries()) {
```

Comment:
Should reconnect in event loop (asynchronously) rather than occupy the thread.

Reply:
fixed